lib/ui/recommendation-row.js

```diff
@@ -30,32 +35,52 @@ RecommendationRow.prototype = {
     recommendation.
     */
     let row = this;
+
+    if (!this.resultType) {
+      return null;
+    }
+
     let renderMethods = [
       row.renderFavicon,
       row.renderImage,
       row.renderContent,
       row.renderLabel
     ];
     let elem = createElement(this.doc, 'hbox', 'container');
+    elem.setAttribute('data-result-type', row.resultType);
     renderMethods.forEach(method => {
       elem.appendChild(method(row));
     });
     return elem;
   },
 
+  getResultType: function(row) {
+  /*
+  Determines and returns a TYPES property indicating the result type.
+  */
+    if (row._get('enhancements.tld')) {
+      return TYPES.TLD;
+    } else if (row._get('enhancements.wikipedia')) {
+      return TYPES.WIKIPEDIA;
+    }
+    return null;
+  },
+
   renderFavicon: function(row) {
     /*
     Creates and returns an <image> element representing the favicon. If none is
     available, or if the URL returned from the server errors, will use
     Firefox's default favicon.
     */
-    let favicon = row._get('enhancements.favicon.url');
     let defaultFavicon = 'chrome://mozapps/skin/places/defaultFavicon@2x.png';
-    let elem = createElement(row.doc, 'image', 'favicon');
+    let wikipedia = 'chrome://universalsearch-skin/content/wikipedia-dark.svg';
+    let favicon = (row.resultType == TYPES.WIKIPEDIA ? wikipedia :
+      (row._get('enhancements.favicon.url') || defaultFavicon));
```

Comment:
I think it'd be best to convert this to the more verbose, clearer style on lines 212-216.